core/src/consensus.rs

```diff
@@ -406,7 +468,16 @@ impl Tower {
     ) -> SwitchForkDecision {
         self.last_voted_slot()
             .map(|last_voted_slot| {
-                let last_vote_ancestors = ancestors.get(&last_voted_slot).unwrap();
+                let last_vote_ancestors = ancestors.get(&last_voted_slot).unwrap_or_else(|| {
```

Comment:
Test that `self.is_stray_last_vote()` should be checked only if `ancestors.get() -> None`